### ISSUE TITLE
Mac M1 termainal can run, but .app crash imme

### ISSUE THREAD

**Issue by zouy50** [burtonageo/cargo-bundle#167]

**Comment by vonnieda** [comment]
Same problem, did you solve it?

I also find this:

> codesign --verify --verbose ../target/release/bundle/osx/Example.app
../target/release/bundle/osx/Example.app: code has no resources but signature indicates they must be present

And I seem to get that regardless of whether I included resources or not.

**Comment by benwr** [comment]
An m1 app bundled using cargo-bundle in October, which used to work, started crashing some time in the last few months. Still working on figuring out why.

**Comment by max-huster** [comment]
Are you trying to access a ./ file? i had the same issiue when opening a log.txt file in the current dir

**Comment by mzdk100** [comment]
Yes, I have resolved this issue. The reason is that the log file was being operated on, and the current directory is not writable when running as an app. Simply changing to a writable path (such as the Application Support directory) solves the problem. For this purpose, I designed a crate called [cache-dir](https://github.com/mzdk100/cache-dir)

**Comment by Mechinsam** [comment]
It's because for some reason macos doesnt look in the Resources folder for.... well..... resources.
The way i got around this was by adding a new function to get the file path from the resources folder with a given filename. its very hacky and jank but it works:

```rust
use std::env::consts::OS;
use std::env;

pub fn get_file_from_absolute_path(file: &str) -> String
{
	
	match OS
	{
		"macos" =>
		{
			let exe_path = env::current_exe().unwrap();
			let abs_filepath = exe_path.parent().unwrap().parent().unwrap().display().to_string() + &format!("/Resources/{}", file);
		
			return abs_filepath;
		}
		_ =>
		{
			let abs_filepath = format!("fatty");

			return abs_filepath;
		}
	}
}
```

**Comment by LinusU** [comment]
@Mechinsam I would recommend using the core_foundation crate instead, I'm doing something like:

```rust
fn resources_root() -> Option<PathBuf> {
    if std::env::var_os("CARGO").is_some() {
        return Some(PathBuf::from(std::env::var_os("CARGO_MANIFEST_DIR")?));
    }

    // TODO: support for other platforms
    #[cfg(target_os = "macos")]
    {
        let bundle = core_foundation::bundle::CFBundle::main_bundle();
        let bundle_path = bundle.path()?;
        let resources_path = bundle.resources_path()?;
        Some(bundle_path.join(resources_path))
    }
    #[cfg(not(any(target_os = "macos")))]
    None
}
```

example: https://github.com/LinusU/rustic-yellow/blob/4a195f50bc2b01f06eb5298895a84ed64aa9b641/src/game/mod.rs#L17-L32

**Comment by ethanuppal** [comment]
> [<img alt="" width="16" height="16" src="https://avatars.githubusercontent.com/u/78374707?u=e2842b0ba89f2f008d147294ee10b2bff2d5f0a0&amp;v=4&amp;size=48">@Mechinsam](https://github.com/Mechinsam) I would recommend using the core_foundation crate instead, I'm doing something like:
> 
> fn resources_root() -> Option<PathBuf> {
>     if std::env::var_os("CARGO").is_some() {
>         return Some(PathBuf::from(std::env::var_os("CARGO_MANIFEST_DIR")?));
>     }
> 
>     // TODO: support for other platforms
>     #[cfg(target_os = "macos")]
>     {
>         let bundle = core_foundation::bundle::CFBundle::main_bundle();
>         let bundle_path = bundle.path()?;
>         let resources_path = bundle.resources_path()?;
>         Some(bundle_path.join(resources_path))
>     }
>     #[cfg(not(any(target_os = "macos")))]
>     None
> }
> example: [LinusU/rustic-yellow@`4a195f5`/src/game/mod.rs#L17-L32](https://github.com/LinusU/rustic-yellow/blob/4a195f50bc2b01f06eb5298895a84ed64aa9b641/src/game/mod.rs#L17-L32)

It would be nice if
1. `cargo-bundle` would set certain compilation flags to indicate that the code was being compiled by `cargo-bundle`
2. `cargo-bundle` offered a library for resolving the path to a resource, allowing the same app to be run
  - from the command line
  - as a `.app` bundle
  - as any other app bundle

(1 would be necessary so a library could even distinguish these possibilities)
